compiler/circle-opselector/driver/Driver.cpp

```diff
@@ -153,6 +155,107 @@ std::vector<std::string> split_name_input(const std::string &str)
   return ::split_into_vector(str, ',');
 }
 
+std::unique_ptr<loco::Graph> make_graph(const std::vector<const luci::CircleNode *> nodes)
+{
+  auto graph = loco::make_graph();
+
+  luci::CloneContext ctx;
+  // clone nodes
+  for (const auto &n : nodes)
+  {
+    auto clone = luci::clone_node(n, graph.get());
+    ctx.emplace(n, clone);
+  }
+  // set graph input
+  for (const auto &n : nodes)
+  {
+    for (uint32_t i = 0; i < n->arity(); i++)
+    {
+      auto arg = n->arg(i);
+      auto input_node = dynamic_cast<luci::CircleNode *>(arg);
+      auto ctx_it = ctx.find(input_node);
+      // check if the node already has been cloned
+      if (ctx_it != ctx.end())
+        continue;
+      // the node isn't graph input if it is an other node's input
+      if (std::find(nodes.begin(), nodes.end(), arg) != nodes.end())
+        continue;
+      auto circle_const = dynamic_cast<luci::CircleConst *>(arg);
+      if (circle_const == nullptr)
+      {
+        // circle input
+        auto circle_input = graph->nodes()->create<luci::CircleInput>();
```

Comment:
how is `NOT CircleConst` sure `IS CircleInput` ?

Reply:
Sorry, incorrectly read the code -_-;

Reply:
@seanshpark Let me explain overall logic.

Parameter `nodes` are selected nodes from user that will be another new graph.

1. First, I cloned each nodes.
```cxx
// clone nodes
  for (const auto &n : nodes)
  {
    auto clone = luci::clone_node(n, graph.get());
    ctx.emplace(n, clone);
  }
```
2. Next, list candidates that could be a graph input.

We may check below conditions to know who is graph input.
- Graph input is a virtual node, which means it wasn't cloned at first step.
```cxx
if (ctx_it != ctx.end())
        continue;
```
- Graph input is not others' input.
```cxx
if (std::find(nodes.begin(), nodes.end(), arg) != nodes.end())
        continue;
```

3. All the nodes that went through above conditions can be a graph input now except one thing; const.

For example, say the node is `CircleConv2D`. it has three inputs - `input`, `filter`, `bias`.

And, only `input` should be a graph input. `filter` and `bias` will be just `CircleConst` nodes.